plugins/shared-react/src/utils/pipeline-utils/pipeline-utils.ts

```diff
@@ -0,0 +1,264 @@
+import { cloneDeep } from 'lodash';
+
+import {
+  cancelledColor,
+  failureColor,
+  pendingColor,
+  runningColor,
+  skippedColor,
+  successColor,
+} from '../../consts';
+import {
+  ComputedStatus,
+  PipelineRunKind,
+  StatusMessage,
+  SucceedConditionReason,
+  TaskRunKind,
+  TaskStatusTypes,
+} from '../../types';
+
+export const getRunStatusColor = (status: string): StatusMessage => {
+  switch (status) {
+    case ComputedStatus.Succeeded:
+      return { message: 'Succeeded', color: successColor };
+    case ComputedStatus.Failed:
+      return { message: 'Failed', color: failureColor };
+    case ComputedStatus.FailedToStart:
+      return {
+        message: 'PipelineRun failed to start',
+        color: failureColor,
+      };
+    case ComputedStatus.Running:
+    case ComputedStatus['In Progress']:
+      return { message: 'Running', color: runningColor };
+
+    case ComputedStatus.Skipped:
+      return { message: 'Skipped', color: skippedColor };
+    case ComputedStatus.Cancelled:
+      return { message: 'Cancelled', color: cancelledColor };
+    case ComputedStatus.Cancelling:
+      return { message: 'Cancelling', color: cancelledColor };
+    case ComputedStatus.Idle:
+    case ComputedStatus.Pending:
+      return { message: 'Pending', color: pendingColor };
+    default:
+      return {
+        message: 'PipelineRun not started yet',
+        color: pendingColor,
+      };
+  }
+};
+
+const getLatestRunBasedOnCreationTimestamp = (
+  runs: PipelineRunKind[],
+  latestRun: PipelineRunKind,
+) => {
```

Comment:
this looks confusing. if latestRun is already there, why do we need to sort it out? and if it needs sorting then we just need `runs` and we have `getLatestRunNotBasedOnCreationTimestamp` which looks similar